kindelia_core/src/node.rs

```diff
@@ -1272,8 +1303,9 @@ impl<C: ProtoComm, S: BlockStorage> Node<C, S> {
         handle_ans_err("GetBlocks", tx.send(infos));
       }
       NodeRequest::GetBlock { hash, tx } => {
-        let info = self.get_block_info(&hash);
-        handle_ans_err("GetBlock", tx.send(info));
+        let info =
+          self.get_block_info(&hash).expect("should obtain block info");
+        handle_ans_err("GetBlock", tx.send(Some(info)));
```

Comment:
I'm not sure I understand what is being asked for.  Nothing can "return an expect" because expect either unwraps a value or panics. 

`get_block_info()` previously could panic in several places and now returns a Result<_, BlockError> instead.  Line 1301 is inside `handle_request() which does not return any Result.  It also could panic previously because it calls `get_block_info()`.  It still can panic in the same place, except now the panic will have a nicely formatted BlockError plus a message from expect().

So the net effect of this change is that we have moved the panic from a lower level to a higher level and added helpful error context.   If/when `handle_request` is refactored to return a Result, then the error can propagate even higher.... end goal of this iterative process being no panics at all.

Now, it *could* be that `handle_request` can simply log/warn about the BlockError and return in either or both of the cases.  I haven't tried to analyze if that is valid behavior or not.  If you think it is good, I'm happy to make that change.  I was just going with the conservative approach of keeping existing logic the same.

Please let me know if I've missed your point...

Reply:
I think the desired behavior would be to take that `Result` returned from `get_block_info` and pass it to `tx.send` instead of an `Option` (which is now always returning `Some` in this case, so has no reason to exist). So the api could return more appropriate results, being able to configure a status code and error message, in case an `Err` is sent by `tx.send`, which was done before, in an archaic way in case of an `Option::None` return.

With this new `expect` I also believe that someone looking for a block that does not exist will cause the api thread to stop with a panic.

My question was just if you think this would be the scope of this PR.

Reply:
Oh, I see.   I think that would be a good improvement.   You could add a commit to this PR, or could merge this, then make a separate one, however you prefer.  I'm unsure how large of a change this would be.   or if you are too busy? I can look into it, please let me know.

> With this new expect I also believe that someone looking for a block that does not exist will cause the api thread to stop with a panic.

good catch.  I didn't really notice before there is a change of behavior because previously get_block_info() could return None due to the `?` on the first line, whereas subsequent lines call expect().

> let block = self.block.get(hash)?;

Reply:
> On second thought, I think the best thing will be to change the result of get_block_info() to <Result<Option<BlockInfo>, BlockError> and remove the BlockError::InvalidBlock variant.

I agree with this, I can do it now and I'll merge after that.

Reply:
It's done in ad52117, I'll wait for your review and if everything is ok I'll merge

Reply:
the change you made had a small bug in the `get_longest_chain` function: it was checking `bhash != zero_hash()` instead of `bhash == zero_hash()`. I analyzed and ended up removing this condition, the `zero_hash()` is not added to the map of included blocks, so it wouldn't pass the condition of the `while` block anyway